scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryToRepositoryDtoMapper.java

```diff
@@ -72,6 +75,13 @@ RepositoryDto createDto(Repository repository) {
     if (RepositoryPermissions.modify(repository).isPermitted()) {
       linksBuilder.single(link("update", resourceLinks.repository().update(repository.getNamespace(), repository.getName())));
     }
+    if (RepositoryPermissions.rename(repository).isPermitted()) {
+      if (scmConfiguration.getNamespaceStrategy().equals("CustomNamespaceStrategy")) {
```

Comment:
The namespace strategies can be implemented by plugins. There may be stragegies, where you can choose from a list. It should be possible for such strategies to change the namespace, too.
Maybe the interface `NamespaceStrategy` needs an additional method like `canBeChanged` or so (by default returning `false`).

Reply:
Sonar has something to say, yoo (though I don't know how to handle the exception hierarchie depth)
https://sonarcloud.io/project/issues?id=sonia.scm%3Ascm&pullRequest=1218&resolved=false&types=CODE_SMELL

Reply:
Fix it. Thanks!